src/IO/Importers/Actions/ReadVolumeData.hpp

```diff
@@ -533,45 +545,57 @@ struct ReadAllVolumeDataAndDistribute {
         } else {
           source_domain = deserialize<Domain<Dim>>(serialized_domain->data());
         }
-        // Reconstruct functions of time from volume data file
-        if (source_domain_functions_of_time.empty() and
-            alg::any_of(source_domain->blocks(), [](const auto& block) {
-              return block.is_time_dependent();
-            })) {
-          const std::optional<std::vector<char>> serialized_functions_of_time =
-              volume_file.get_functions_of_time(observation_id);
-          if (not serialized_functions_of_time.has_value()) {
-            ERROR_NO_TRACE("No domain functions of time found in file '"
-                           << file_name << volume_file.subfile_path()
-                           << "'. The functions of time are needed for "
-                              "interpolating with time-dependent maps.");
-          }
-          source_domain_functions_of_time = deserialize<std::unordered_map<
-              std::string,
-              std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>>(
-              serialized_functions_of_time->data());
+      } else {
+        if (elements_match) {
+          Parallel::printf(
+              "WARNING: No serialized domain found in file. "
+              "Verification that elements in the source and target domain "
+              "match will be skipped.\n");
```

Comment:
I'm not sure we should allow this. Is there any scenario where somebody has an h5 file with volume data in the exact configuration of domain, but without the serialized domain?

Reply:
Yes if they generate the file by providing data at given coordinates. It's important not to babysit users too much, i.e. allow the "I know what I'm doing is correct, just let me do it" scenario. Serializing a domain into the H5 file just to enable a check is too strict.